From 49951f72cfda2f66ad8e703064f0ec9288294ae8 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 06 Jul 2022 17:24:59 +0300
Subject: [PATCH] [SW-94367] gaudi2: additional cfg in sris mode

This patch adds change for clearing pll short lock enable registers
as part of sris programming sequence recommended by arch team.

Change-Id: I605389deeffda0ad430426b13bbbf10b74568261
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index d19180d..36fd052 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -257,6 +257,14 @@
 	regmap_writel(0x0, mmD0_PCIE_PHY_PIPE_LANE_CMN_REFCLK_MODE);
 }
 
+void pcie_phy_set_pll_short_lock(void)
+{
+	regmap_writel(0x0, mmD0_PCIE_PHY_PHYX_MPLL_SHORT_LOCK_EN_0);
+	regmap_writel(0x0, mmD0_PCIE_PHY_PHYX_MPLL_SHORT_LOCK_EN_1);
+	regmap_writel(0x0, mmD0_PCIE_PHY_PHYX_MPLL_SHORT_LOCK_EN_2);
+	regmap_writel(0x0, mmD0_PCIE_PHY_PHYX_MPLL_SHORT_LOCK_EN_3);
+}
+
 uint8_t get_sris_mode_from_strap(void)
 {
 	uint32_t boot_strap;
diff --git a/common/pcie.c b/common/pcie.c
index b5f5770..0ed6d4c 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -74,6 +74,7 @@
 __weak void pcie_set_pcie_device_vendor_id(void) {}
 __weak void pcie_config_boot_termination(void) {}
 __weak void pcie_phy_set_lane_refclk_mode(void) {}
+__weak void pcie_phy_set_pll_short_lock(void) {}
 
 void pcie_config(void)
 {
@@ -98,6 +99,7 @@
 		if (get_sris_mode_from_strap()) {
 			pcie_phy_set_lane_refclk_mode();
 			regmap_writel(1, mmPIPE_RX_SRIS_MODE_EN);
+			pcie_phy_set_pll_short_lock();
 			regmap_writel(1, mmAPP_SRIS_MODE);
 		}
 
